ts/client/src/client.ts

```diff
@@ -3374,64 +3372,230 @@ export class MangoClient {
     return await this.sendAndConfirmTransactionForGroup(group, [ix]);
   }
 
-  /**
-   * Example:
-   * For a stop loss on SOL, assuming SOL/USDC pair
-   * priceLowerLimit - e.g.
-   *
-   * @param group
-   * @param account
-   * @param sellMintPk - would be SOL mint
-   * @param priceLowerLimit - priceLowerLimit would be greater than priceUpperLimit e.g. if SOL is at 25$, then priceLowerLimit could be 22$
-   * @param buyMintPk - would be USDC mint
-   * @param priceUpperLimit - priceLowerLimit would be greater than priceUpperLimit e.g. if SOL is at 25$, then priceUpperLimit could be 0$
-   * @param maxSell - max ui amount of tokens to sell, e.g. account.getTokenBalanceUi(solBank)
-   * @param pricePremiumFraction - premium in % the liquidator earns for executing this stop loss, this can be the slippage usually found for a particular size plus some buffer
-   * @param expiryTimestamp - is epoch in seconds at which this stop loss should expire, set null if you want it to never expire
-   * @returns
-   */
-  public async tokenConditionalSwapStopLoss(
+  public async tcsTakeProfitOnDeposit(
     group: Group,
     account: MangoAccount,
-    sellMintPk: PublicKey,
-    priceLowerLimit: number,
-    buyMintPk: PublicKey | null,
-    priceUpperLimit: number | null,
-    maxSell: number | null,
-    pricePremiumFraction: number | null,
+    sellBank: Bank,
+    buyBank: Bank,
+    thresholdPriceUi: number,
+    thresholdPriceInSellPerBuyToken: boolean,
+    maxBuyUi: number | null,
+    maxSellUi: number | null,
+    pricePremium: number | null,
+    allowCreatingDeposits: boolean | null,
+    allowCreatingBorrows: boolean | null,
     expiryTimestamp: number | null,
   ): Promise<TransactionSignature> {
-    const buyBank: Bank = group.getFirstBankByMint(buyMintPk ?? USDC_MINT);
-    const sellBank: Bank = group.getFirstBankByMint(sellMintPk);
+    if (maxSellUi == null && account.getTokenBalanceUi(sellBank) < 0) {
```

Comment:
why check only when maxSellUi is null?

Reply:
because in this function, normally, we wouldnt care about maxBuyUi, we are doing a take profit, so we would want to this on entire deposit

Reply:
If we take profit/stop loss a deposit, I think we should always check that the sell bank has a positive deposit - no matter if maxSell is specified or not.